src/features/MentorPage/ListCard/ListCard.tsx

```diff
@@ -5,41 +5,47 @@ import ListCardLanguages from './ListCardLanguages';
 import ListCardSkills from './ListCardSkills';
 import ListCardStory from './ListCardStory';
 import OpenCardButton from './OpenCardButton';
-import * as cssVariables from '../../../components/CommonTextStyles/variables';
+import * as cssVariables from '../../../components/variables';
+import { handleSetVisibleCardProps } from '../MentorPage';
 
 /**
- * Div to hold mentor cards. Needs a parameter added
- * That hold the info for cards that need to be showed.
+ * A single card with mentor info on mentor page card listing
  */
 
 type Props = {
-  setVisibleCard: (mentorCardData: ListCardProps) => void;
-  mentorData: ListCardProps;
+  setVisibleCard: ({
+    shouldShowMentorCard,
+    mentorCardData,
+  }: handleSetVisibleCardProps) => void;
+  mentorCardData: ListCardProps;
 };
 
-const ListCard = ({ setVisibleCard, mentorData }: Props) => {
-  const age = new Date().getFullYear() - mentorData.mentor.birthYear;
-  return (
-    <ListCardElement>
-      <ListCardHeader
-        name={mentorData.mentor.displayName}
-        age={age}
-        region={mentorData.mentor.region}
-        isAvailable={mentorData.isLoggedIn}
-        isNewMentor={mentorData.isNewMentor}
-        message={mentorData.contactMessage}
-      />
-      <CardContent>
-        <ListCardStory story={mentorData.mentor.story} />
-        <ListCardLanguages languages={mentorData.mentor.languages} />
-        <ListCardSkills skills={mentorData.mentor.skills} />
-        <OpenCardButton
-          setVisibleCard={setVisibleCard}
-          mentorData={mentorData}
+const ListCard = ({ setVisibleCard, mentorCardData }: Props) => {
+  if (mentorCardData !== undefined) {
+    const age = new Date().getFullYear() - mentorCardData.mentor.birthYear;
+    return (
+      <ListCardElement>
+        <ListCardHeader
+          name={mentorCardData.mentor.displayName}
+          age={age}
+          region={mentorCardData.mentor.region}
+          isAvailable={mentorCardData.isLoggedIn}
+          isNewMentor={mentorCardData.isNewMentor}
+          message={mentorCardData.contactMessage}
         />
-      </CardContent>
-    </ListCardElement>
-  );
+        <CardContent>
+          <ListCardStory story={mentorCardData.mentor.story} />
+          <ListCardLanguages languages={mentorCardData.mentor.languages} />
+          <ListCardSkills skills={mentorCardData.mentor.skills} />
+          <OpenCardButton
+            setVisibleCard={setVisibleCard}
+            mentorCardData={mentorCardData}
+          />
+        </CardContent>
+      </ListCardElement>
+    );
+  }
+  return <></>;
```

Comment:
I find it easier to read if we reverse the condition here. 
Also we dont need to return a fragment.
So =>

![image](https://user-images.githubusercontent.com/34128180/183646693-7a5929b9-aa2c-4a8f-b3f4-f2be3142731a.png)